### PR TITLE
Williams update

### PR DESCRIPTION
**Pull request by wJohnwarne**
## Description

The following changes are made to move this repo more towards a one-stop-shop for everything needed for a developer to onboard themselves to 'the delta engineering team'.

**The BE Guide**
Shall contain navigation to all things relevant to a backend developer at the delta studio:
- Templates
- Commonly used tools. (this section may also be useful for the PO team and pre-sales team to see what tools are available to us that can be used in projects)
- Style guides
- and so on. (Could add 'Concepts' section where add good links on specific topics instead of dumping them in the slack readings channel.)

**Lessons and Notes**
In the commonly used tools section, there is a Notes/Lessons section. This would take the place of the slack lessons channel in order to keep a more ordered (and therefore useful) recorded of lessons learnt.

**How to use**
- To add a lesson simply make a 'Lesson PR' and add the info in the relevant section.
- If you use a new tool, add a section about it in the 'Commonly used tools' section.

### CODE REVIEW

File: back-end/Readme.MD [hunk 1]
```diff
@@ -0,0 +1,73 @@
+# The Delta Studio Backend Guide
+
+*This guide documents eveything needed to work within The Delta Backend team, including tools, templates, style guides, commonly used packages and 3rd party software.*
+
+## Templates
+
+*Commonly used templates developed and maintained by employees of The Delta Studio.*
+
+- [Dotnet core 3.1 C# API](https://github.com/The-Delta-Studio/tds-dotnet-core-3.1-api-template)
+- [Dotnet core 2.1 C# API](https://github.com/The-Delta-Studio/tds-api-template)
```
Comment (RupertLeslie-Delta) [inline back-end/Readme.MD:10]: Pretty sure this is 2.2?

File: back-end/Readme.MD [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+Swagger is a API documentation handler that helps generate usefull API documentation dirrectly from the API code.
+
+##### Aditional
+
+- Implementation [guide](https://docs.microsoft.com/en-us/samples/aspnet/aspnetcore.docs/getstarted-swashbuckle-aspnetcore/?tabs=visual-studio) for effective API documentation.
+
+
+#### [Sendgrid](https://sendgrid.com/)
+
+Integratable email distrobution service enable our APIs to send emails with dynamically populated data.
```
Comment (RupertLeslie-Delta) [inline back-end/Readme.MD:38]: *distribution

File: back-end/Readme.MD [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+*This guide documents eveything needed to work within The Delta Backend team, including tools, templates, style guides, commonly used packages and 3rd party software.*
+
+## Templates
+
+*Commonly used templates developed and maintained by employees of The Delta Studio.*
+
+- [Dotnet core 3.1 C# API](https://github.com/The-Delta-Studio/tds-dotnet-core-3.1-api-template)
+- [Dotnet core 2.1 C# API](https://github.com/The-Delta-Studio/tds-api-template)
+- [Dotnet core AWS Lambda function](https://github.com/The-Delta-Studio/tds-func-lambda-di-ef-template)
```
Comment (deltaDirk) [inline back-end/Readme.MD:11]: There are 2 templates for Lambda functions. Maybe just mention this one is with DI and link the other please :)

File: back-end/Readme.MD [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+
+#### [Sendgrid](https://sendgrid.com/)
+
+Integratable email distrobution service enable our APIs to send emails with dynamically populated data. 
+
+##### Notes/Lessons
+
+1. **DELAYED SENDING:** Emails can sometime be quite delayed which may cause issues.
+2. **ACCOUNT WARMUP:** Accounts need to be 'warmed up'. This is done by sending multible small batches of emails over a period of days. The purpose of this is to prevent the 'account' being blocked by email providers (when larger volumes of emails are sent) causing emails to not reach the intended recipients.
```
Comment (deltaDirk) [inline back-end/Readme.MD:43]: Spelling mistake on multiple

File: back-end/Readme.MD [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+A framework that allows tasks or processes to be offloaded to background proccessing. Support recurring, delayed, fire-and-forget, and more.
+
+##### Aditional
+
+- C# [package](https://www.nuget.org/packages/Hangfire/)
+
+#### [Retool](https://docs.retool.com/docs)
+
+A tool that can be used to quickly and easerly create a Frontend that can interact with an API or database. Often used to create admin portals.
```
Comment (deltaDirk) [inline back-end/Readme.MD:59]: Spelling mistake on easerly

File: back-end/Readme.MD (hunk 1 above)
Comment (wJohnwarne) [reply back-end/Readme.MD:10]: Ya just checked and you're correct, shall change 👍

File: back-end/example-files/YomaDockerFile [hunk 6]
```diff
@@ -0,0 +1,28 @@
+FROM mcr.microsoft.com/dotnet/core/sdk:3.1 AS builder
+
+WORKDIR /build
+COPY ./YomaApiCore/YomaApiCore.csproj ./
```
Comment (Roleen20) [inline back-end/example-files/YomaDockerFile:4]: maybe we can rename this to projectFolderLocation/projectName.csproj 
and the rest below referencing YomaApiCore

Comment (wJohnwarne) [reply back-end/example-files/YomaDockerFile:4]: Ya that's a good point, I'll change that now 👍

**Pull request merged**